src/main/java/com/subscriptionmanager/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,210 @@
+                .setYear(expiryDate.getYear())
+                .build())
+            .build())
+        .build();
+  }
+
+  @Override
+  public RenewUserSubscriptionResponse renewUserSubscription(
+      final RenewUserSubscriptionRequest request) {
+
+    final Map<String, String> parentResourceValueMap =
+        validationService.validateAndExtractResourceValueMap(request.getParent(), USERS);
+
+    final User user = userRepositoryService.findById(parentResourceValueMap.get(USERS));
+
+    final Map<String, String> nameResourceValueMap =
+        validationService.validateAndExtractResourceValueMap(request.getName(), SUBSCRIPTIONS);
+
+    final Subscription subscription =
+        subscriptionRepositoryService.findById(nameResourceValueMap.get(SUBSCRIPTIONS));
+
+    final UserSubscription userSubscription = userSubscriptionRepository
+        .findByUserAndSubscription(user, subscription)
+        .orElseThrow(() -> ResourceNotFoundException.builder()
+            .resourceName(USER_SUBSCRIPTION)
+            .fieldName("Mapping")
+            .fieldValue(subscription.toString() + user.toString())
+            .build()
+        );
+
+    final LocalDate expiryDate = userSubscription.getExpiryDate()
+        .toLocalDate()
+        .plusDays(subscription.getValidity());
+
+    userSubscription.setExpiryDate(Date.valueOf(expiryDate));
+
+    userSubscriptionRepository.save(userSubscription);
+
+    return RenewUserSubscriptionResponse.newBuilder()
+        .setUserSubscription(com.subscriptionmanager.v1.proto.UserSubscription.newBuilder()
+            .setName(SUBSCRIPTIONS.concat("/").concat(subscription.getId()))
+            .setDisplayName(subscription.getName())
+            .setExpiryDate(com.google.type.Date.newBuilder()
+                .setDay(expiryDate.getDayOfMonth())
+                .setMonth(expiryDate.getMonthValue())
+                .setYear(expiryDate.getYear())
+                .build())
+            .build())
+        .build();
+  }
+
+  @Override
+  public RemoveUserSubscriptionResponse removeUserSubscription(
+      final RemoveUserSubscriptionRequest request) {
+
+    final Map<String, String> variableValueMap =
+        validationService.validateAndExtractResourceValueMap(request.getName(), USERS,
+            SUBSCRIPTIONS);
+
+    final User user = userRepositoryService.findById(variableValueMap.get(USERS));
+
+    final Subscription subscription =
+        subscriptionRepositoryService.findById(variableValueMap.get(SUBSCRIPTIONS));
+
+    final UserSubscription userSubscription = userSubscriptionRepository
+        .findByUserAndSubscription(user, subscription)
+        .orElseThrow(() -> ResourceNotFoundException.builder()
+            .resourceName(USER_SUBSCRIPTION)
+            .fieldValue(subscription.toString() + user.toString())
+            .build()
+        );
+
+    final LocalDate expiryDate = userSubscription.getExpiryDate().toLocalDate();
+
+    userSubscriptionRepository.delete(userSubscription);
```

Comment:
delete mostly represent hard delete in internal services. 
disable would be a better word.